src/transformers/models/swin/configuration_swin.py

```diff
@@ -0,0 +1,143 @@
+# coding=utf-8
+# Copyright The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" Swin Transformer model configuration """
+
+from ...configuration_utils import PretrainedConfig
+from ...utils import logging
+from torch import nn
+
+
+logger = logging.get_logger(__name__)
+
+SWIN_PRETRAINED_CONFIG_ARCHIVE_MAP = {
+    "microsoft/swin-base": "https://huggingface.co/microsoft/swin-base patch4-window12-384/resolve/main/config.json",
+    # See all Swin models at https://huggingface.co/models?filter=swin
+}
+
+
+class SwinConfig(PretrainedConfig):
+    r"""
+    This is the configuration class to store the configuration of a [`~SwinModel`].
+    It is used to instantiate an Swin model according to the specified arguments, defining the model
+    architecture. Instantiating a configuration with the defaults will yield a similar configuration to that of
+    the Swin [swin-base](https://huggingface.co/swin-base) architecture.
+
+    Configuration objects inherit from  [`PretrainedConfig`] and can be used
+    to control the model outputs. Read the documentation from  [`PretrainedConfig`]
+    for more information.
+
+
+    Args:
+        image_size (`int`, *optional*, defaults to 224):
+            Size (resolution) of the input images.
+        patch_size (`int`, *optional*, defaults to 4):
+            Size of patches.
+        num_channels (`int`, *optional*, defaults to 3):
+            Number of channels in input images.
+        num_labels (`int`, *optional*, defaults to 1000):
+            Number of classes for classification head
```

Comment:
We usually don't document `num_labels`, as it's already documented for `PretrainedConfig`.